nomos-services/blend/src/backends/libp2p.rs

```diff
@@ -16,18 +16,13 @@ use nomos_libp2p::{secret_key_serde, NetworkBehaviour};
 use overwatch_rs::overwatch::handle::OverwatchHandle;
 use rand::RngCore;
 use serde::{Deserialize, Serialize};
-use tokio::{
-    sync::{broadcast, mpsc},
-    task::JoinHandle,
-};
+use tokio::sync::{broadcast, mpsc};
 use tokio_stream::wrappers::BroadcastStream;
 
 use super::BlendBackend;
 
 /// A blend backend that uses the libp2p network stack.
 pub struct Libp2pBlendBackend {
-    #[expect(dead_code)]
-    task: JoinHandle<()>,
```

Comment:
I think this is wrong. In this case the task should be wrapped in an abortable and get aborted when stopping the service if possible. That is why it probably was there.

Reply:
Tests kept passing and @youngjoon-lee said it was a leftover from a previous refactoring. I am personally not sure why it's there. Should I revert the change for now?

Reply:
Actuall after thinking about this a bit, it wasn't used anyway here, so it was just something for future reference. I would say we remove it here, since the lint enforces it, and then we can re-add it AND make use of it once we want to use it. Sounds good?

Reply:
@danielSanchezQ Ah, now I can recall the history. Thanks for reminding. 

When I first implemented it, I saved JoinHandle for later use for [Abortable](https://docs.rs/futures/latest/futures/future/struct.Abortable.html). At that time, I referred to DA network backend code. Later, [Abortable was added to DA code](https://github.com/logos-co/nomos-node/pull/853/files#diff-618f063bf12ec5d0d7faae32b79afd2d86d2a6db75d46b8abe0d4fb78489afd2L74), but I didn't change this Blend code yet. I should have added TODO comment (as we did in DA). Sorry for the confusion.

First of all, I agree to remove this for making clippy happy.I will open issue right now, review if adding Abortable is necessary and work on it.